WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductSelectorViewModel.swift

```diff
@@ -178,10 +187,20 @@ final class ProductSelectorViewModel: ObservableObject {
         guard let selectedProduct = products.first(where: { $0.productID == productID }) else {
             return
         }
-        if let onProductSelected = onProductSelected {
-            onProductSelected(selectedProduct)
+
+        if shouldToggleItemOnSelection {
+            // Product multi-selection (new)
+            if let onProductSelected = onProductSelected {
+                toggleSelection(productID: productID)
+                onProductSelected(selectedProduct)
+            }
         } else {
-            toggleSelection(productID: productID)
+            // Product single-selection (legacy)
+            if let onProductSelected = onProductSelected {
+                onProductSelected(selectedProduct)
+            } else {
+                toggleSelection(productID: productID)
+            }
```

Comment:
💡 This is more of a nit: I'd prefer not calling the new logic multi-selection, since we have multi-selection before and it just behaves in a slightly different way. How about structuring these checks in another way to make it clearer to see the differences:

```Swift
        if shouldToggleItemOnSelection, let onProductSelected {
            // toggles the item and then triggers the selection
            toggleSelection(productID: productID)
            onProductSelected(selectedProduct)
        } else if let onProductSelected {
            // the selector supports single selection only, so the closure should be triggered and the work is done.
            onProductSelected(selectedProduct)
        } else {
            // no closure for single item selection is available; toggles the selection only for multiple selections.
            toggleSelection(productID: productID)
        }
```

Reply:
Looking at the suggested change above, I think maybe renaming `shouldToggleItemOnSelection` into `supportsMultipleSelections` would be more precise. WDYT?

Reply:
> How about structuring these checks in another way to make it clearer to see the differences:

Looks great, an less verbose than my approach, changed on 976e6e2f4d8e59055c7c34f7ef1dc49ce1d90f50

> renaming

I can't seem to figure out a proper name for this flag, as should not only convey multi-selection is allowed but also that there's logic happening after each selection (unlike coupons). That said, `supportsMultipleSelections` definitely sounds better than `shouldToggleItemOnSelection`, I can re-think variable this once we move to M1.1, as the logic changes slightly as well. Changed on e8bdc191ff7d4dcb20439fa12758ca1bc3c1508a